component/src/main/java/org/wso2/rule/validator/document/Document.java

```diff
@@ -169,15 +170,22 @@ private List<LintResult> lintNode(String path, Rule rule) throws InvalidRulesetE
         }
         for (RuleThen then : rule.then) {
             List<LintTarget> lintTargets = getLintTargets(node, then);
-
             for (LintTarget target : lintTargets) {
                 List<String> parentPath = splitJsonPath(path);
                 parentPath.addAll(target.jsonPath);
                 String targetPath = LintTarget.getPathString(parentPath);
                 target.jsonPath = parentPath;
                 FunctionResult result = then.lintFunction.execute(target);
-                results.add(new LintResult(result.passed, targetPath, rule,
-                        rule.message == null ? result.message : rule.message));
+                MessagePlaceholder placeholder = new MessagePlaceholder(
```

Comment:
Shall we move this object creation to inside the if clause? This object `placeholder` is only used if the condition is true. Otherwise it's not used at all. So if the condition is false, the creation of this object is a waste of resources.